feg/cloud/go/services/feg_relay/gw_to_feg_relay/servicers/session_proxy_relay.go

```diff
@@ -19,85 +19,117 @@ import (
 	"github.com/golang/glog"
 
 	"magma/lte/cloud/go/protos"
+	"magma/orc8r/cloud/go/services/dispatcher/gateway_registry"
 )
 
 // SessionProxyServer implementation
 //
-// Notify OCS/PCRF of new session and return rules associated with subscriber
+// CreateSession Notifies OCS/PCRF of new session and return rules associated with subscriber
 // along with credits for each rule
 func (s *RelayRouter) CreateSession(
 	ctx context.Context, r *protos.CreateSessionRequest) (*protos.CreateSessionResponse, error) {
 
 	// CreateSession's SID is just IMSI with "IMSI" prefix which findServingNHFeg() should remove
-	client, ctx, cancel, err := s.getSessionProxyClient(ctx, r.GetCommonContext().GetSid().GetId())
+	client, ctx, cancel, err := s.getSessionControllerProxyClient(ctx, r.GetCommonContext().GetRatType(), r.GetCommonContext().GetSid().GetId())
 	if err != nil {
 		return nil, err
 	}
 	defer cancel()
 	return client.CreateSession(ctx, r)
 }
 
-// Updates OCS/PCRF with used credit and terminations from gateway
+// UpdateSession Updates OCS/PCRF with used credit and terminations from gateway
 func (s *RelayRouter) UpdateSession(
 	ctx context.Context, r *protos.UpdateSessionRequest) (*protos.UpdateSessionResponse, error) {
 
-	// Group requests by PLMNID
-	// Use the longest possible PLMN ID (6), in the worst case we'll fragment requests more then needed, but the routing
-	// will still be logically correct
-	reqMap := map[string]*protos.UpdateSessionRequest{}
+	// Group requests by [PLMNID][GwServiceType]
+	// - PLMNID: Use the longest possible PLMN ID (6), in the worst case we'll fragment requests more then needed,
+	//   but the routing will still be logically correct
+	// - GwServiceType: Each request will be sent to either Session_Proxy or for N7_N40_proxy
+	reqMap := map[string]map[gateway_registry.GwServiceType]*protos.UpdateSessionRequest{}
```

Comment:
why do you need map of maps here (the map of proxy types)? It seems that you are not using the proxy type map, just initializing it and then using GetRatType() instead anyway. Maps are relatively expensive and it complicates your code.

Reply:
Hey @emakeev  I am trying to remember why but I think the idea is to group the request. 

So for example, sessiond will send all the updates in one message
```
Update A
	PLMN 1
	RAT 4G
Update B  
	PLMN 1
	RAT 4G
Update C 
	PLMN 2
	RAT 5G
```

The result of the `map[PLMN][RAT]` should be 
```
[1][4g] = [A, B]
[2][4g] = [C]
```

Note that [A,B] should be sent  from feg relay to session_proxy in one single message which contains both updates


Norte it is true that RAT is not being used later anymore

Reply:
Hi @emakeev , to echo @uri200 , UpdateSessionRequest from sessiond populates updates of 4G and 5G. Map of Maps is used to maintain  plmn to RATType mappings i.e., [PLMN][RATType].  And then relays the request to SessionProxy or N7N40Proxy of FeG.

Reply:
Hey @anusha-konda , can you forward this pr to @GANESH-WAVELABS  or other members on Wavelabs so they can review it too please?